packages/services/ds/src/workflows/server/pegboard/mod.rs

```diff
@@ -646,27 +695,57 @@ async fn reschedule_actor(
 	})
 	.await?;
 
-	let res = setup(ctx, &input, false, new_image_id).await;
-	match ctx.catch_unrecoverable(res)? {
-		Ok(_actor_id) => {}
-		Err(err) => {
-			tracing::error!(?err, "unrecoverable reschedule");
+	let actor_setup = setup(ctx, &input, SetupCtx::Reschedule { new_image_id }).await?;
+
+	// Waits for the actor to be ready (or destroyed) and automatically retries if failed to allocate.
+	ctx.repeat(|ctx| {
+		let input = input.clone();
+		let actor_setup = actor_setup.clone();
+
+		async move {
+			spawn_actor(ctx, &input, &actor_setup).await?;
+
+			match ctx.listen::<Init>().await? {
+				Init::ActorStateUpdate(sig) => match sig.state {
+					pp::ActorState::Allocated {
+						client_id: _client_id,
+					} => return Ok(Loop::Break(None)),
+					pp::ActorState::FailedToAllocate => {
+						// Get and increment retry count
+						let retry_count = ctx
+							.activity(UpdateRescheduleRetryInput {
+								server_id: input.server_id,
+								reset: false,
+							})
+							.await?;
 
-			ctx.workflow(destroy::Input {
-				server_id: input.server_id,
-				override_kill_timeout_ms: None,
-				signal_actor: false,
-			})
-			.output()
-			.await?;
+						// Determine next backoff sleep duration
+						let mut backoff = rivet_util::Backoff::new_at(
+							8,
+							None,
+							BASE_RETRY_TIMEOUT_MS,
+							500,
+							retry_count.try_into()?,
+						);
+						let next = backoff.step().expect("should not have max retry");
 
-			// Throw the original error from the setup activities
-			return Err(err);
-		}
-	};
+						// Sleep for backoff
+						ctx.sleep_until(next).await?;
```

Comment:
I think this is still wrong. We talked about also adding backoff for an actor that crashes, so shouldn't the backoff be executed at the start of reschedule_actor? Like `if !first_reschedule { /* do backoff */ }`

Context: https://rivet-gg.slack.com/archives/D02R3KVNKHN/p1732672061669669?thread_ts=1732671085.517149&cid=D02R3KVNKHN

Reply:
Actors always retry immediately the first time, then backoff kicks in: https://rivet-gg.slack.com/archives/D02R3KVNKHN/p1732672128678809?thread_ts=1732671085.517149&cid=D02R3KVNKHN